### DISCUSSION TITLE
zoom not fully including the requested bounds

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#1087] [Q&A] [answered]
While working on [this c:geo issue](https://github.com/cgeo/cgeo/issues/15033) I noticed, that the viewport that results from a `mMap.animator().animateTo(bounds)` does not fully include the requested viewport (which leads to caches and/or waypoints being outside the visible viewport). The viewport is not completely wrong, it's the right area etc., but there's missing a bit around the edges. 

Happens while working with a Mapsforge offline map, have not tested yet with other map types.

(Cannot name the exact differences but only test visually currently, due to `getBoundBox()` giving weird results, see other discussion.)

**Comment by devemux86** [comment]
Markers have (pixel) size, so can increase the `BoundingBox` by some percentage, e.g. `BoundingBox.extendMargin(1.2f)`.

Can also use `MapPosition` after calculating it with `MapPosition.setByBoundingBox`.

**Comment by moving-bits** [comment, marked as answer]
Ok... (*blush on*) it looks like I haven't set the "include waypoints in zoom" option in c:geo, thus the viewport gets calculated for caches only, leaving some waypoints outside the visible area. In conjunction with the other issue (which may or may not be a timing issue) I did not notice that I hadn't set the config option... Sorry for wasting your time.
